libs/blocks/mas-autoblock/mas-autoblock.js

```diff
@@ -19,10 +23,27 @@ export function getTagName(el) {
 }
 
 export async function createCard(el, fragment) {
+  // add <mas-commerce-service>
+  const servicePromise = initService();
+  const timeoutPromise = new Promise((resolve) => {
+    setTimeout(() => resolve(false), MAS_AUTOBLOCK_TIMEOUT);
+  });
+  let success = await Promise.race([servicePromise, timeoutPromise]);
+  if (!success) {
+    throw new Error('Failed to initialize mas commerce service');
+  }
```

Comment:
shouldn't we rather move all that logic within initService? making servicePromise the race promise?

Reply:
maybe yes, maybe no.
right now it gives consumer flexibility to set timeout or not to set.
maybe for ccd slow machine its fine for service to initialize 10 seconds